packages/client-sdk-nodejs/src/cache-client.ts

```diff
@@ -47,6 +49,12 @@ export class CacheClient extends AbstractCacheClient implements ICacheClient {
       configuration: configuration,
     };
 
+    const numConcurrentRequests = configuration
+      .getTransportStrategy()
+      .getGrpcConfig()
+      .getConcurrentRequestsLimit();
+    CacheClient.semaphore = new Semaphore(numConcurrentRequests);
```

Comment:
Should use [a static initialization block](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks) to instantiate the semaphore since it is a class variable.

As is here, we write a new static semaphore every time an instance of the cache client is created, which would lead to ill defined behavior for already existing clients. There's also a problem as to what happens if I instantiate clients with different limits of concurrent requests... or if I instantiate one client with no concurrent request limit and another with it.